scooter_manipulation/src/go_to_joint_config.cpp

```diff
@@ -0,0 +1,112 @@
+#include <functional>
+#include <memory>
+#include <thread>
+
+#include "scooter_interfaces/action/go_to_joint_config.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_action/rclcpp_action.hpp"
+#include "rclcpp_components/register_node_macro.hpp"
+
+#include <moveit/move_group_interface/move_group_interface.h>
+#include <moveit/planning_scene_interface/planning_scene_interface.h>
+
+// I'm not doing the visibility control thing because this will never be run on Windows ever for any reason
+
+#define PLANNING_GROUP "ur_manipulator"
+
+namespace go_to_joint_config_cpp {
+/**
+ *  ROS action server that plans and executes a trajectory to a given joint configuration for the UR5. Uses message of
+ *  type `scooter_interfaces/GoToJointConfig`.
+ */
+class GTJCActionServer : public rclcpp::Node {
+ public:
+  using GTJC = scooter_interfaces::action::GoToJointConfig;
+  using GoalHandleGTJC = rclcpp_action::ServerGoalHandle<GTJC>;
+
+  explicit GTJCActionServer(std::shared_ptr<rclcpp::Node> move_group_node,
+                            const rclcpp::NodeOptions &options = rclcpp::NodeOptions())
+      : Node("go_to_joint_config", options) {
+    using namespace std::placeholders;
+
+    move_group_node_ = move_group_node;
+
+    // initialize move group
+    move_group_ = std::make_shared<moveit::planning_interface::MoveGroupInterface>(move_group_node_, PLANNING_GROUP);
+
+    // initialize action server callbacks
+    this->action_server_ = rclcpp_action::create_server<GTJC>(
+        this,
+        "go_to_joint_config",
+        std::bind(&GTJCActionServer::handle_goal, this, _1, _2),
+        std::bind(&GTJCActionServer::handle_cancel, this, _1),
+        std::bind(&GTJCActionServer::handle_accepted, this, _1)
+    );
+  }
+
+ private:
+  // MoveIt stuff
+  rclcpp::NodeOptions node_options_;
+  rclcpp::Node::SharedPtr move_group_node_;
+  moveit::planning_interface::MoveGroupInterfacePtr move_group_;
+
+  // action server stuff
+  rclcpp_action::Server<GTJC>::SharedPtr action_server_;
+
+  rclcpp_action::GoalResponse handle_goal(
+      const rclcpp_action::GoalUUID &uuid,
+      std::shared_ptr<const GTJC::Goal> goal
+  ) {
+    RCLCPP_INFO(this->get_logger(), "Received goal request");
+    (void) uuid;
+    (void) goal;
+    return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
+  }
+
+  rclcpp_action::CancelResponse handle_cancel(
+      const std::shared_ptr<GoalHandleGTJC> goal_handle
+  ) {
+    RCLCPP_INFO(this->get_logger(), "Received request to cancel goal");
+    (void) goal_handle;
```

Comment:
What does this void cast do here?

Reply:
The void cast is entirely to suppress the unused parameter warning which unfortunately happens because each of these callbacks needs to have this exact signature even when the parameter is unnecessary.